pyproject.toml

```diff
@@ -31,9 +31,9 @@ ops = "^2.14.1"
 # data_platform_libs/v0/upgrade.py
 # grafana_agent/v0/cos_agent.py requires pydantic <2
 pydantic = "^1.10.17, <2"
-# tls_certificates_interface/v1/tls_certificates.py
-cryptography = "^42.0.8"
-jsonschema = "^4.21.1"
+# tls_certificates_interface/v3/tls_certificates.py
+cryptography = "*"
+jsonschema = "*"
```

Comment:
This is too broad, we need to pin

Reply:
@Mehdi-Bendriss that was agreed between me and @carlcsaposs-canonical. That is b/c `tls_certificates` lib sets its own `PYDEPS` to the latest available versions for both. So, if we pin, we are effectively using a different version then what they are testing themselves.

Reply:
btw, that is only updated if we push the `poetry.lock` file.

Reply:
@Mehdi-Bendriss, the wildcard pins are in the `charm-libs` dependency group that is aimed to follow what is expected by `PYDEPS` in various charm libraries, with the actual version pins being in the main dependency group. Poetry would pick the more restrictive version pin, if they are not in conflict, so `^42.0.8` + `*` would have been the same as just `^42.0.8`.

The idea is that we should be able to automate collection at some point and resolve what the charm expects and what the libraries expect.

Reply:
Thank you all for the context!